workflow/controller/operator.go

```diff
@@ -1224,6 +1224,17 @@ func (woc *wfOperationCtx) podReconciliation(ctx context.Context) (error, bool)
 				woc.updated = true
 			}
 			woc.markNodePhase(node.Name, wfv1.NodeError, "pod deleted")
```

Comment:
this should probably also use `markNodeError` while we're making this modification
```suggestion
			woc.markNodeError(node.Name, errors.New("pod deleted"))
```